### PR TITLE
Improved interface help interaction

### PR DESCRIPTION
**Pull request by peterfpeterson**
**Description of work.**

Rather than having all of the interfaces know how to create the various urls for help, consolidate that functionality into the central function for displaying help.

Lots more commits than what actually happened. The changeset is correct.

**To test:**

Bring up the help for "DGS planner" in the various operation modes.

*There is no associated issue.*

The sibling into `ornl-next` is #30787.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/gui_helper.py [hunk 1]
```diff
@@ -83,24 +115,40 @@ def closeEvent(self, event):
         event.accept()
     '''
     try:
+        # try using built-in help in mantid
         import mantidqt
-        mantidqt.interfacemanager.InterfaceManager().showCustomInterfaceHelp(mantidplot_name)
+        mantidqt.interfacemanager.InterfaceManager().showCustomInterfaceHelp(mantidplot_name, section)
```
Comment (AndreiSavici) [inline qt/python/mantidqt/gui_helper.py:120]: This line will always fail. Try
`mantidqt.interfacemanager.InterfaceManager().showCustomInterfaceHelp('/'.join([mantidplot_name, section]))` instead, or you need to pass the parameters down the chain

**Review by AndreiSavici — DISMISSED** [review]
Will need to also check all interfaces, to make sure there are no missing pages. PyChop needs to have the 'direct' section

**Comment by AndreiSavici** [comment]
@martyngigg Can you get some people at ISIS going through all interfaces, see if they can open the correct help page?

**Review by martyngigg — APPROVED** [review]
The ISIS interfaces have been checked and all of them open the correct help page when clicked.

**Pull request merged**